src/object/object_primitive.c

```diff
@@ -12194,11 +12194,25 @@ mr_cmpval_char (DB_VALUE * value1, DB_VALUE * value2, int do_coercion, int total
       return DB_UNK;
     }
 
+
+
   c =
     QSTR_CHAR_COMPARE (collation, string1, (int) DB_GET_STRING_SIZE (value1), string2,
 		       (int) DB_GET_STRING_SIZE (value2));
   c = MR_CMP_RETURN_CODE (c);
 
+/*
+ *  Check if the compare_LT flag set. If so, return -1 for the "%" and "chr(255)*" patterns.
+ */
+
+  if (value2->data.ch.info.compare_LT == true)
```

Comment:
to compare with a max string, we should check string values. it is enough to check the flag.

consider that first value might be "max string".

return DB_LT, DB_GT depending on case.